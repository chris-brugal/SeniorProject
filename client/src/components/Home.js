import React from "react";
import "../styles/LandingPage.css";
import { Typography, Grid, Button } from "@mui/material";
import Typewriter from "typewriter-effect";
import { Link as ScrollLink } from "react-scroll";

const Home = () => {
  return (
    <div id="home" className="home-div">
      <Grid container spacing={1}>
        <Grid
          item
          xs={12}
          md={5}
          container
          justify="center"
          alignItems="center"
        >
          <Typography
            className="main-text"
            variant="h2"
            sx={{
              paddingLeft: {
                xs: "0px",
                md: "30px",
              },
              paddingTop: {
                xs: "20px",
                md: "0px",
              },
            }}
          >
            Welcome to Spotify Central, begin
            <div className="typew">
              <Typewriter
                skipAddStyles
                options={{
                  strings: [
                    "Listening.",
                    "Analyzing.",
                    "Creating.",
                    "Exploring.",
                  ],
                  autoStart: true,
                  loop: true,
                }}
              />
            </div>
            <ScrollLink
              activeClass="active"
              to="About Us"
              spy={true}
              smooth={true}
              duration={500}
            >
              <Button variant="contained">Get Started</Button>
            </ScrollLink>
          </Typography>
        </Grid>
        <Grid item xs={12} md={7} container justifyContent="flex-end">
          <img
            src="/images/player.jpg"
            className="main-img"
            alt="Music Player"
          />
        </Grid>
      </Grid>
    </div>
  );
};

export default Home;
