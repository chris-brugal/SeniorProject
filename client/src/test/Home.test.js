import React from 'react';
import Home from '../components/Home';
import renderer from "react-test-renderer";

describe("snapshot testing", () => {

test("snapshot testing for Home component", () => {
    const comp = renderer.create(<Home />).toJSON()
    expect(comp).toMatchSnapshot()
})
})