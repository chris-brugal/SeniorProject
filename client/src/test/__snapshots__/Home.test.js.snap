// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot testing for Home component 1`] = `
<div
  className="home-div"
  id="home"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-sc7rel-MuiGrid-root"
      justify="center"
    >
      <h2
        className="MuiTypography-root MuiTypography-h2 main-text css-155ym27-MuiTypography-root"
      >
        Welcome to Spotify Central, begin
        <div
          className="typew"
        >
          <div
            className="Typewriter"
            data-testid="typewriter-wrapper"
          />
        </div>
        <a
          onClick={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Get Started
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </h2>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 css-19mjduq-MuiGrid-root"
    >
      <img
        alt="Music Player"
        className="main-img"
        src="/images/player.jpg"
      />
    </div>
  </div>
</div>
`;
